streams/src/main/java/org/apache/kafka/streams/state/internals/Segment.java

```diff
@@ -48,4 +48,17 @@ public void openDB(final ProcessorContext context) {
     public String toString() {
         return "Segment(id=" + id + ", name=" + name() + ")";
     }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (getClass() != obj.getClass()) return false;
+
+        Segment segment = (Segment) obj;
+        return Long.compare(id, segment.id) > 1;
```

Comment:
Left this out of my previous review.

This needs to change as well as it will return false for two `Segment` instances with the same `id`.  

Also, this statement will always evaluate to false as `Long.compare(x, y)` only returns `-1`, `0`, or `1`.